esp/bindings/SOAP/Platform/soapbind.cpp

```diff
@@ -313,7 +336,12 @@ void CSoapRequestBinding::post(const char *proxy, const char* url, IRpcResponseB
         soapclient.setReadTimeoutSecs(readTimeoutSecs_);
     if (mtls_secret_.length())
         soapclient.setMtlsSecretName(mtls_secret_);
-    if (client_cert_.length() || ca_certs_.length() || accept_self_signed_)
+
+    if (client_cert_buf_.length() || ca_certs_buf_.length())
+        soapclient.setSecureSocketConfig(createSecClientConfigBuf(client_cert_buf_, client_priv_key_buf_, ca_certs_buf_, accept_self_signed_));
```

Comment:
may be simpler as a single function that also takes the buffer parameters - it would remove some duplicate code and would also work with a mixuture of buffers and files (although I can't see that being used)